wp-content/themes/goonj-crm/functions.php

```diff
@@ -191,7 +191,30 @@ function goonj_handle_user_identification_form() {
 		}
 
 		// If we are here, then it means the user exists as an inducted volunteer.
-		wp_redirect(get_home_url() . "/collection-camp-form/#?source_contact_id=" . $contact['id'] );
+		// Fetch the most recent collection camp activity based on the creation date
+		$collectionCampResult = civicrm_api3('Activity', 'get', [
+			'sequential' => 1,
+			'contact_id' => $contact['id'],
+			'activity_type_id' => ['IN' => [61]], // ID for "Collection Camp Intent"
+			'status_id' => ['IN' => [10]], // Status ID for "Under Authorization"
+			'order_by' => 'created_date DESC',
+			'limit' => 1,
+		]);
+
+		// Recent camp data
+		$recentCamp = end($collectionCampResult['values']);
```

Comment:
In APIv4 there is a method called `first` that can be used for this. [See example usage](https://github.com/ColoredCow/goonj/blob/70047e1dca7cb0d1d3347fa1740516486114adba/wp-content/themes/goonj-crm/functions.php#L292)
```suggestion
		$recentCamp = $result->first();
```